conf/config-schema.json

```diff
@@ -0,0 +1,16 @@
+{
+    "type": "object",
+    "properties": {
+        "root": { "type": "boolean" },
+        "globals": { "type": ["object", "array"] },
+        "global": { "type": ["object", "array"] },
```

Comment:
Ok, after digging deeper into this I think I have a better understanding of what's going on. Sorry for the long post.

As @mysticatea pointed out, `global` object nor `globals` array are valid in the config.

There is one case of `global` array in the tests, but it is a false negative. So we can definitely remove the array option.

The Config object itself accepts `globals` as an array and converts it to an object. https://github.com/eslint/eslint/blob/8733e6a852ea9b49cef26650357598ca8c2c2a1f/lib/config.js#L214
This is used for the command line options. e.g `--global foo --global bar` so we need to keep that

So the only question is whether we should remove `global` object.

`eslint` `prepareConfig` has a conditional to use `global` if `globals` is falsy. 
https://github.com/eslint/eslint/blob/8733e6a852ea9b49cef26650357598ca8c2c2a1f/lib/eslint.js#L453
This code was added 3 years ago. I think `globals` was added later, and `global` was kept for backwards compatibility.

I think in a Config object `globals` is always truthy, so this could be removed. 
https://github.com/eslint/eslint/blob/8733e6a852ea9b49cef26650357598ca8c2c2a1f/lib/config.js#L214

The only gotcha is in RuleTester we are validating a regular object, not built through Config. So it happily accepts `global` or `globals`
and passes it to eslint.verify.

There are tests from 2 years ago (v.1.0.0 days) validating this behavior: https://github.com/eslint/eslint/blame/8733e6a852ea9b49cef26650357598ca8c2c2a1f/tests/lib/testers/rule-tester.js#L479

However the jsdoc in verify mentions it expects `globals` not `global`.
https://github.com/eslint/eslint/blob/8733e6a852ea9b49cef26650357598ca8c2c2a1f/lib/eslint.js#L742

So, to the best of my knowledge, `global` is only used by our tests and could be removed. It might break some people tests if they are using it too. If we decide to keep it, we should skip it or special case it, so the validation doesn't fail.